### PR TITLE
389 component checkbox

### PR DESCRIPTION
**Pull request by arnau-rius**
## Create Checkbox component

**TASK**: [#389](https://github.com/adevinta/spark/issues/389)

### Description, Motivation and Context
This PR adds a initial iteration of the Checkbox

- Added initial documentation
- Added initial tests without testing implementation details
- Added all the needed styles and motion

### Types of changes
- [ ] 🛠️ Tool
- [ ] 🪲 Bug fix (non-breaking change which fixes an issue)
- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🧾 Documentation
- [ ] 📷 Demo
- [ ] 🧪 Test
- [ ] 🧠 Refactor
- [ ] 💄 Styles

### Screenshots - Animations

https://user-images.githubusercontent.com/32937662/226363876-4b448da3-7360-4a11-9b51-a82a14aca19a.mov

### CODE REVIEW

**Review by andresin87 — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -32,6 +32,7 @@
     "url": "git@github.com:adevinta/spark.git"
   },
   "dependencies": {
+    "@radix-ui/react-checkbox": "1.0.3",
```
Comment (andresz1) [inline package.json:35]: This should be inside the component

File: packages/components/checkbox/src/Checkbox.test.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+import { describe, expect, it, vi } from 'vitest'
+
+import { Checkbox } from './Checkbox'
+
+describe('Checkbox', () => {
+  it('should render', () => {
+    render(
+      <>
+        <Checkbox id={'c1'} />
+        <label htmlFor={'c1'}>Accept terms and conditions</label>
```
Comment (andresz1) [inline packages/components/checkbox/src/Checkbox.test.tsx:12]:
```
<label htmlFor={'c1'}>
```
➡️
```
<label htmlFor="c1">
```

File: packages/components/checkbox/src/Checkbox.test.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+      </>
+    )
+
+    expect(screen.getByText('Accept terms and conditions')).toBeInTheDocument()
+    expect(screen.getByRole('checkbox')).not.toBeChecked()
```
Comment (andresz1) [inline packages/components/checkbox/src/Checkbox.test.tsx:17]: You are not testing that `Accept terms and conditions` is a label
```
expect(screen.getByRole('checkbox', {name: 'Accept terms and conditions' })).not.toBeChecked()
```

File: packages/components/checkbox/package.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    "access": "public"
+  },
+  "main": "./dist/index.js",
+  "module": "./dist/index.mjs",
+  "types": "./dist/index.d.ts",
+  "scripts": {
+    "build": "vite build"
+  },
+  "peerDependencies": {
+    "@radix-ui/react-checkbox": "1.0.3"
```
Comment (andresz1) [inline packages/components/checkbox/package.json:15]: What about `cva` dependency ?

File: .storybook/preview.cjs [hunk 5]
```diff
@@ -1,7 +1,6 @@
 import '../src/tailwind.css'
 
 export const parameters = {
-  actions: { argTypesRegex: '^on[A-Z].*' },
```
[the remark this answers is not in the document]
Comment (acd02) [reply .storybook/preview.cjs:4]: Hm, even though git blame points to me, this line wasn't actually my work 😅
All I did was add the Tailwind import

cc @Powerplex ?

**Review by andresz1 — APPROVED** [review]

File: .storybook/preview.cjs (hunk 5 above)
[the remark this answers is not in the document]
Comment (solygambas) [reply .storybook/preview.cjs:4]: Hi, here is more info about actions: https://storybook.js.org/docs/react/essentials/actions

[the remark this answers is not in the document]
Comment (arnau-rius) [reply .storybook/preview.cjs:4]: What's happening is that is overriding the prop `onCheckedChange` from the Checkbox component and is breaking the stories. Is there a way to disable this just for this prop?

File: package.json [hunk 6]
```diff
@@ -32,6 +32,7 @@
     "url": "git@github.com:adevinta/spark.git"
   },
   "dependencies": {
+    "@radix-ui/react-checkbox": "^1.0.3",
```
Comment (andresin87) [inline package.json:35]: remove

**Review by andresin87 — APPROVED** [review]

**Pull request merged**